index.js

```diff
@@ -328,6 +336,11 @@ function prettyFormat(val, opts) {
     opts = normalizeOptions(opts);
   }
 
+  const output = opts.highlight
+    ? (val, type) => chalk[opts.theme[type]](val)
+    : val => val
+  ;
```

Comment:
It might be faster to build up an object once with the functions and to use the style strings themselves. Looks like chalk uses getters for some reason: https://github.com/chalk/chalk/blob/master/index.js#L25-L27

```js
const highlighter = {};

Object.keys(opts.theme).forEach(key => {
  if (opts.highlight) {
    let styleName = opts.theme[key];
    let {open, close} = chalk.styles[styleName];
    highlighter[key] = val => open + val + close;
  } else {
    highlighter[key] = val => val;
  }
});
```

Then pass that around instead. Hopefully it will be a bit faster. Also you could probably use the base ansi-styles library and install less.